### ISSUE TITLE
round-robin LB if multiple `backendRef`s are present on an `HTTPRoute`

### ISSUE THREAD

**Issue by mflendrich** [Kong/kubernetes-ingress-controller#2173]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem Statement

If an `HTTPRoute` has multiple `backendRef` entries, Kong should assume a round-robin LB strategy.

Today multiple backendRefs on an HTTPRoute fail validation.

### Proposed Solution

Implement `backendRef` 1-1 mapping to upstreams. Change KIC to allow more than 1 backendRef on an HTTPRoute.

### Prerequisites

- [ ] #2209 

### Additional information

Follow-ups for this ticket:

- [x] #2174
- [x] #2175

### Acceptance Criteria

- [ ] HTTPRoutes with multiple `backendRef`s no longer fail validation
- [ ] HTTPRoutes with multiple `backendRef`s result in an upstream being created in Kong for each backendRef

**Comment by shaneutt** [comment]
After several rounds of review and some attempts to solve this problem I'm finding there is significant technical debt within the `internal/parser` package that makes it difficult (though not impossible) to add support for multiple backendRefs services on top of the existing code. I'm marking this blocked temporarily while I focus on checking in with teammates and seeing if we might want to hold off on shoehorning in support in favor of adding the requirements to #2130 and resolving it more holistically.

**Comment by shaneutt** [comment]
After discussing with the team we've decided that we need to spike the options available to implement multiple backend Kubernetes services support in parser, #2209 is now a prerequisite for this issue and as such we're going to move it to milestone 2 scope as we agreed this does not need to block milestone 1.

**Comment by mflendrich** [comment]
blocked on #2295

**Comment by shaneutt** [comment]
closing as a duplicate of #2166, and the AC has been moved to the description there.
